kubespawner/objects.py

```diff
@@ -61,6 +61,10 @@
 
 from .utils import get_k8s_model, update_k8s_model
 
+SERVICE_DNS_PATTERN = re.compile(
```

Comment:
An explanation here, or a reference to where its explained, seems very relevant. This influences KubeSpawner code as well and I care deeply that this project is possible to maintain long term - having comments like that can be essential.